pfcpiface/telemetry.go

```diff
@@ -89,5 +108,20 @@ func (uc *upfCollector) summaryLatencyJitter(ch chan<- prometheus.Metric) {
 func setupProm(upf *upf) {
 	uc := newUpfCollector(upf)
 	prometheus.MustRegister(uc)
+	promStats.pfcpMessages = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "pfcp_messages_total",
+		Help: "Counter for incoming and outgoing PFCP messages",
+	}, []string{"node_id", "message_type", "direction", "result"})
+	promStats.activeSessionCnt = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "active_current_sessions_total",
+		Help: "The total number of Users with active sessions in UPF",
+	}, []string{"node_id"})
```

Comment:
```
package main

type pfcpStats struct {
	messages       *prometheus.CounterVec
	activeSessions *prometheus.GaugeVec
}

var pfcpStats *pfcpStats

func newPFCPStats() *pfcpStats {
	return &pfcpStats{
		messages: prometheus.NewCounterVec(prometheus.CounterOpts{
			Name: "pfcp_messages_total",
			Help: "Counter for incoming and outgoing PFCP messages",
		}, []string{"node_id", "message_type", "direction", "result"}),

		activeSessions: prometheus.NewGaugeVec(prometheus.GaugeOpts{
			Name: "active_sessions_total",
			Help: "The total number of Users with active sessions in UPF",
		}, []string{"node_id"}),
	}
}

func (ps *pfcpStats) Describe(ch chan<- *prometheus.Desc) {
	ch <- ps.messages
	ch <- ps.activeSessions
}

func setupProm(upf *upf) {
	pfcpStats = newPFCPStats()
	prometheus.MustRegister(pfcpStats)
}
```

Reply:
@badhrinathpa do we need 2 nodeID related labels for messages? localNodeID and remoteNodeID? Basically when metrics from more than 1 UPF is being aggregated we need a key to differentiate right? And one label to differentiate mutliple SMFs on 1 UPF

Reply:
@badhrinathpa actually never mind we can expect the higher level k8s/prometheus integration to add the extra label that would differentiate the node from which the metric is being collected.

Reply:
adding describe functions needs us to define a custom collector. so not doing that for now since we are using the prometheus provided metrics itself.